File: MemorablePlaces/app/src/main/java/com/descifrador/memorableplaces/MainActivity.java
package com.descifrador.memorableplaces;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import com.google.android.gms.maps.model.LatLng;

import java.util.ArrayList;
import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    static ArrayList<String> places = new ArrayList<>();
    static ArrayList<LatLng> loca = new ArrayList<>();

    ListView memorableList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        memorableList = findViewById(R.id.memorableList);

        if(places.isEmpty()){
            places.add("Add memorable place");
            loca.add(new LatLng(0.0,0.0));
        }

        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<>(this,android.R.layout.simple_list_item_1,places);
        memorableList.setAdapter(arrayAdapter);

        memorableList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            Intent intent = new Intent(getApplicationContext(),SecondActivity.class);
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                if(position==0){
                    intent.putExtra("position",position);
                    startActivity(intent);
                }
                else
                {
                    intent.putExtra("position",position);
                    startActivity(intent);
                    Toast.makeText(MainActivity.this, places.get(position)+" "+loca.get(position).toString(),Toast.LENGTH_SHORT).show();
                    //Toast.makeText(MainActivity.this, places.get(position), Toast.LENGTH_SHORT).show();
                }
            }
        });

    }

    @Override
    protected void onResume() {
        super.onResume();


    }
}

File: MVPModelDemo/app/src/main/java/com/descifrador/mvpmodeldemo/MainActivity.kt
package com.descifrador.mvpmodeldemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.Toast
import com.descifrador.mvpmodeldemo.presenter.ILoginPresenter
import com.descifrador.mvpmodeldemo.presenter.LoginPresenter
import com.descifrador.mvpmodeldemo.view.ILoginView
import kotlinx.android.synthetic.main.activity_main.*
import kotlin.math.log

class MainActivity : AppCompatActivity() , ILoginView {

    private lateinit var loginPresenter : ILoginPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        loginPresenter = LoginPresenter(this)

        val btnClickMe = findViewById<Button>(R.id.loginButton)

        btnClickMe.setOnClickListener{
            loginPresenter.onLogin(emailText.text.toString(),PassText.text.toString())
        }

    }

    override fun onLoginResult(msg: String) {
        Toast.makeText(this,msg,Toast.LENGTH_SHORT).show()
    }
}